hw/boards/mta1-usb-v1/ch552_fw/main.c

```diff
@@ -572,46 +584,76 @@ main()
     UEP0_T_LEN = 0;
     UEP1_T_LEN = 0;                                                       //Pre-use send length must be cleared
     UEP2_T_LEN = 0;                                                       //Pre-use send length must be cleared
-    
+
+    // Enable GPIO debugging on p1.4 and p1.5
+    // gpio_init();
+    // gpio_unset(0x10);
+    // gpio_unset(0x20);
+
     while(1)
     {
         if(UsbConfig)
         {
             if(USBByteCount)   // USB receiving endpoint has data
             {
-                CH554UART1SendByte(Ep2Buffer[USBBufOutPoint++]);
-                recievedData[USBBufOutPoint] = Ep2Buffer[USBBufOutPoint];
-                USBByteCount--;
+                memcpy(USB_output_buffer, Ep2Buffer, USBByteCount);
```

Comment:
dehanj and I discussed wrapping this section inside USB_IE=0/USB_IE=1 to disable interrupts temporarily to protect Ep2Buffer and USBByteCount from being updated while we were copying data from it. It turns out that we're already answering NAK on USB at this time, so that's unneccessary. So it's OK by me to leave as is.